channelz/service/func_linux.go

```diff
@@ -37,7 +37,7 @@ func sockoptToProto(skopts *channelz.SocketOptionData) []*channelzpb.SocketOptio
 	if skopts.Linger != nil {
 		opts = append(opts, &channelzpb.SocketOption{
 			Name: "SO_LINGER",
-			Additional: testutils.MarshalAny(&channelzpb.SocketOptionLinger{
+			Additional: utils.MustMarshalAny(&channelzpb.SocketOptionLinger{
```

Comment:
This is a good catch. But we shouldn't be using a function that can `panic` in the first place. My suggestion would be to instead the `ptypes.MarshalAny()` here and throw a warning log if it fails, and don't append to `opts` in that case.

Reply:
And if we do that, we can let the `testutils` package as is. Or if we really want to move things around, `utils` is not a very informative package name. Please see: https://google.github.io/styleguide/go/decisions#package-names. Specifically see the line which says:
Avoid uninformative package names like util, utility, common, helper, and so on. See more about [so-called “utility packages”](https://google.github.io/styleguide/go/best-practices#util-packages).

I would suggest having two packages instead:
1. `internal/marshalutil` with the function `MustMarshalAny`
2. `internal/parseutil` with function `MustParseURL`.

What are your thoughts on this @dfawley ?

Reply:
IMO let's leave things mostly as they are and avoid churn.  The existing `testutils` function is for testing convenience and can panic, while the production code should do the same thing but error instead of panic.  There's no value in wrappers for production code since we'd just be calling the one parse/marshal function and returning its error.

Reply:
> The existing testutils function is for testing convenience and can panic, while the production code should do the same thing but error instead of panic.

In this case I would suggest to address what I mentioned in PR description:
> NOTE: I would suggest adding a CI check (in the GitHub workflow) to prevent future introductions of the same dependency.

Otherwise this problem (accidentally using a `testutils` function in production code) might happen again.

I will try to refactor the existing non-test code to not use `testutils`.

Reply:
I have reworked the PR to not use any new package and added a regression test to try to cover the existing bug (panic), PTAL.

Reply:
> NOTE: I would suggest adding a CI check (in the GitHub workflow) to prevent future introductions of the same dependency.

Yes, something like this would be a good idea.  However, it might be hard to configure correctly.  We could probably leverage some google-internal infrastructure instead, which would be pretty trivial to enable, to catch this upon doing our weekly import at least.

Reply:
This is what I came up in a couple minutes:
```bash
#!/bin/bash

set -e

FOUND=0
for PKG in $(go list ./...); do
	if echo "$PKG" | grep -qF test; then
		echo "Skipping package $PKG" 1>&2
		continue
	fi
	if go list -deps $PKG | grep -q ^testing$; then
		echo "== Package $PKG depends on 'testing'" 1>&2;
		FOUND=$((FOUND+1))
	fi
done

if [ $FOUND -ne 0 ]; then
	echo "FAIL: found $FOUND packages depending on 'testing' package" 1>&2
	exit $FOUND
fi

echo "OK: no package depends on 'testing'"
```

But it's not very elegant/future-proof.